File: 数据结构和算法/python/selectionSort.py
# 选择排序的实现
def selectionSort(array):
    i = 0
    while i < len(array):
        j = i
        minNum = array[j]
        minNumIndex = j
        while j < len(array):
            if minNum > array[j]:
                minNum = array[j]
                minNumIndex = j
            j += 1
        array[minNumIndex] = array[i]
        array[i] = minNum
        i += 1
    return array

if __name__ == "__main__":
    arr = selectionSort([7,2,34,9,-1,7])
    print (arr)

File: 数据结构和算法/java/SelectionSort.java
/**
 * 选择排序的实现
 */
public class SelectionSort{
    public int[] sort(int[] array){
        for (int i = 0; i < array.length; i++){
            int minNum = array[i];
            int minIndex = i;
            for (int j = i; j < array.length; j++){
                if (minNum > array[j]){
                    minNum = array[j];
                    minIndex = j;
                }
            }
            array[minIndex] = array[i];
            array[i] = minNum;
        }
        return array;
    }

    public static void main(String[] args) {
        SelectionSort selectionSort = new SelectionSort();
        int[] array = new int[]{7,2,34,9,-1,7};
        int[] sortArray = selectionSort.sort(array);
        for (int num : sortArray){
            System.out.print(num + ", ");
        }
        
    }
}